### PR TITLE
testmap: Run sub-man-cockpit on Fedora 41

### PR DESCRIPTION
**Pull request by martinpitt**
- [x] https://github.com/candlepin/subscription-manager-cockpit/pull/79

### CODE REVIEW

**Comment by martinpitt** [comment]
@ptoscano FYI

**Comment by martinpitt** [comment]
@ptoscano : This fails because sub-man checks out an ancient cockpit lib (based on version 292). This failure happens because Fedora 41 moved to dnf 5, and packagelib.py was fixed accordingly in https://github.com/cockpit-project/cockpit/commit/c71ae28e557ae79210537bd0af8f6fcf5c9ad20c .

This is going to be a huge jump (including the CDP → BiDi transition, which is imperative if you want to test on Firefox), but apparently becomes pressing now.

File: lib/testmap.py [hunk 1]
```diff
@@ -214,9 +214,9 @@ def contexts(image: str, *scenarios: Iterable[str], repo: str | None = None) ->
             'rhel-10-0',
             'fedora-39',
             'fedora-40',
+            'fedora-41',
```
Comment (martinpitt) [inline lib/testmap.py:217]: This needs to be applied to sub-man above as well

**Comment by ptoscano** [comment]
> @ptoscano : This fails because sub-man checks out an ancient cockpit lib (based on version 292). This failure happens because Fedora 41 moved to dnf 5, and packagelib.py was fixed accordingly in [cockpit-project/cockpit@c71ae28](https://github.com/cockpit-project/cockpit/commit/c71ae28e557ae79210537bd0af8f6fcf5c9ad20c) .

I see; we have few occurrences of dnf too:
```
$ PAGER= git grep -nw dnf
.github/workflows/stylish.yml:17:        dnf --setopt install_weak_deps=False install -y \
README.md:22:    $ sudo dnf install gettext nodejs npm make
test/vm.install-sub-man:4:dnf install -y cockpit-ws python3-devel openssl-devel intltool gcc
```
- `stylish.yml` uses the fedora:stable container; once f41 is released then we'll need to adapt to it to dnf5 (may be nothing to do)
- `README.md` doesn't matter
- `vm.install-sub-man` should work fine just fine with dnf5?

> This is going to be a huge jump (including the CDP → BiDi transition, which is imperative if you want to test on Firefox), but apparently becomes pressing now.

I'm not sure what you are talking about; do you have any references/links about it?

**Comment by martinpitt** [comment]
@ptoscano This is the place that checks out cockpit's test/common: https://github.com/candlepin/subscription-manager-cockpit/blob/630d2319a5a638958d04b2d8c84ccc97b382da1e/Makefile#L225  d9426e07a261c182d61163900740fd74d02fe99b is the cockpit commit. The comment "292 + sizzle workaround for Chromium 113" describes it.

**Comment by ptoscano** [comment]
> @ptoscano This is the place that checks out cockpit's test/common: https://github.com/candlepin/subscription-manager-cockpit/blob/630d2319a5a638958d04b2d8c84ccc97b382da1e/Makefile#L225 d9426e07a261c182d61163900740fd74d02fe99b is the cockpit commit. The comment "292 + sizzle workaround for Chromium 113" describes it.

Oh that I know :D I was referring to your comments about the "huge jump" in the version requirement, and which other changes in cockpit may require changes in sub-man-cockpit.

**Comment by martinpitt** [comment]
@ptoscano You don't import pkg/lib, so there shouldn't be node-modules changes (other than that you can drop `chrome-remote-interface`). But it's still a big jump, I'd be surprised if it "just works". If there's something unclear, please feel free to @ me in the sub-man PR.

**Comment by ptoscano** [comment]
Bumping the cockpit test version seemed doable, with a couple of easy changes outside the test themselves:
https://github.com/candlepin/subscription-manager-cockpit/pull/79
it seems to be working fine!

**Comment by martinpitt** [comment]
Yay, all green, thanks @ptoscano ! Welcome to Fedora 41.

**Comment by ptoscano** [comment]
> Yay, all green, thanks @ptoscano ! Welcome to Fedora 41.

nice, thank you!

**Review by tomasmatus — APPROVED** [review]

**Pull request merged**